dev/test/query.ts

```diff
@@ -621,6 +627,38 @@ describe('query interface', () => {
       });
     });
   });
+
+  it('for Query.withConverter()', async () => {
+    const doc = document('documentId', 'author', 'author', 'title', 'post');
+    const overrides: ApiOverride = {
+      commit: (request, options, callback) => {
+        const expectedRequest = set({
+          document: doc,
+        });
+        requestEquals(request, expectedRequest);
+
+        callback(null, writeResult(1));
+      },
+      runQuery: request => {
+        queryEquals(request, fieldFilters('title', 'EQUAL', 'post'));
+        return stream({document: doc, readTime: {seconds: 5, nanos: 6}});
+      },
+    };
+
+    return createInstance(overrides).then(async firestore => {
+      await firestore
+        .collection('collectionId')
+        .doc('documentId')
+        .set({title: 'post', author: 'author'});
+      const posts = await firestore
+        .collection('collectionId')
+        .where('title', '==', 'post')
+        .withConverter(postConverter)
+        .get();
+      expect(posts.size).to.equal(1);
+      expect(posts.docs[0].data()!.byline()).to.equal('post, by author');
```

Comment:
You should be able to remove the bang since queries only return documents that have data (via QueryDocumentSnapshot).